### PR TITLE
Add fluentd examples

### PR DESCRIPTION
**Pull request by zhu733756**
Signed-off-by: zhu733756 <zhu733756@kubesphere.io>

Points:
  -  convert deployment to sts
  -  add fluentd examples

This pr needs to be merged before #192

### CODE REVIEW

File: controllers/fluent_controller_finalizer.go [hunk 1]
```diff
@@ -108,12 +108,15 @@ func (r *FluentdReconciler) mutate(obj client.Object, fd *fluentdv1alpha1.Fluent
...
+	case *appsv1.StatefulSet:
+		expected := operator.MakeStatefulset(*fd)
 
 		return func() error {
 			o.Labels = expected.Labels
-			o.Spec = expected.Spec
+			o.Spec.Replicas = expected.Spec.Replicas
+			o.Spec.Template = expected.Spec.Template
+			o.Spec.UpdateStrategy = expected.Spec.UpdateStrategy
+			o.Spec.VolumeClaimTemplates = expected.Spec.VolumeClaimTemplates
```
Comment (benjaminhuo) [inline controllers/fluent_controller_finalizer.go:119]: sts use `VolumeClaimTemplates`, do we still need `case PersistentVolumeClaim` above?

File: apis/fluentd/v1alpha1/plugins/common/buffer_types.go [hunk 2]
```diff
@@ -163,7 +169,24 @@ func (b *Buffer) Params(_ plugins.SecretLoader) (*params.PluginStore, error) {
 	}
 
 	if b.Path != nil {
-		ps.InsertPairs("path", *b.Path)
+		if strings.HasPrefix(*b.Path, "/buffers") {
```
Comment (zhu733756) [inline apis/fluentd/v1alpha1/plugins/common/buffer_types.go:172]: Buffer PVC mounts with /buffers.

File: controllers/fluent_controller_finalizer.go (hunk 1 above)
Comment (zhu733756) [reply controllers/fluent_controller_finalizer.go:119]: I think no need. I will fix it and have a test.

File: controllers/fluentd_controller.go [hunk 3]
```diff
@@ -197,7 +197,7 @@ func (r *FluentdReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&fluentdv1alpha1.Fluentd{}).
 		Owns(&corev1.ServiceAccount{}).
-		Owns(&appsv1.Deployment{}).
+		Owns(&appsv1.StatefulSet{}).
 		Owns(&corev1.Service{}).
 		Owns(&corev1.PersistentVolumeClaim{}).
```
Comment (benjaminhuo) [inline controllers/fluentd_controller.go:189]: no need to owns `PersistentVolumeClaim ` now?

Comment (zhu733756) [reply controllers/fluentd_controller.go]: Done

**Review by benjaminhuo — APPROVED** [review]

File: apis/fluentd/v1alpha1/fluentd_types.go [hunk 4]
```diff
@@ -34,6 +34,8 @@ type FluentdSpec struct {
 	DisableService bool `json:"disableService,omitempty"`
 	// Numbers of the Fluentd instance
 	Replicas *int32 `json:"replicas,omitempty"`
+	// Numbers of the workers in Fluentd instance
+	Workers  *int32 `json:"workers,omitempty"`
```
Comment (zhu733756) [inline apis/fluentd/v1alpha1/fluentd_types.go:38]: Support multi workers

**Pull request merged**